runtime-modules/project-token/src/tests/patronage.rs

```diff
@@ -0,0 +1,520 @@
+    let expected = balance!(95);
+
+    let token_info = TokenDataBuilder::new_empty()
+        .with_patronage_rate(BlockRate::from_yearly_rate(init_rate, BlocksPerYear::get()))
+        .build();
+    let config = GenesisConfigBuilder::new_empty()
+        .with_token_and_owner(token_id, token_info, owner, init_supply)
+        .build();
+
+    build_test_externalities(config).execute_with(|| {
+        increase_block_number_by(blocks);
+        let _ = <Token as PalletToken<AccountId, Policy, IssuanceParams>>::reduce_patronage_rate_by(
+            token_id, decrement,
+        );
+
+        increase_block_number_by(blocks);
+        let _ = <Token as PalletToken<AccountId, Policy, IssuanceParams>>::reduce_patronage_rate_by(
+            token_id, decrement,
+        );
+
+        // initially zero: now updated
+        assert_eq!(
+            Token::token_info_by_id(token_id)
+                .patronage_info
+                .unclaimed_patronage_tally_amount,
+            expected,
+        );
+    })
+}
+
+#[test]
+fn decrease_patronage_ok_with_event_deposit() {
+    let init_rate = yearly_rate!(50);
+    let token_id = token!(1);
+    let decrement = yearly_rate!(20);
+
+    let params = TokenDataBuilder::new_empty()
+        .with_patronage_rate(BlockRate::from_yearly_rate(init_rate, BlocksPerYear::get()));
+    let config = GenesisConfigBuilder::new_empty()
+        .with_token(token_id, params.build())
+        .build();
+
+    let final_rate = BlockRate::from_yearly_rate(yearly_rate!(30), BlocksPerYear::get())
+        .to_yearly_rate_representation(BlocksPerYear::get());
+
+    build_test_externalities(config).execute_with(|| {
+        let _ = <Token as PalletToken<AccountId, Policy, IssuanceParams>>::reduce_patronage_rate_by(
+            token_id, decrement,
+        );
+
+        last_event_eq!(RawEvent::PatronageRateDecreasedTo(token_id, final_rate));
+    })
+}
+
+#[test]
+fn decrease_patronage_ok_with_new_patronage_rate_correctly_approximated() {
+    let init_rate = yearly_rate!(50);
+    let token_id = token!(1);
+    let decrement = yearly_rate!(20);
+    // hard coded reference value: (.3/52594921) * 1quintillion
+    let expected = BlockRate(Perquintill::from_parts(5703972822));
+
+    let params = TokenDataBuilder::new_empty()
+        .with_patronage_rate(BlockRate::from_yearly_rate(init_rate, BlocksPerYear::get()));
+    let config = GenesisConfigBuilder::new_empty()
+        .with_token(token_id, params.build())
+        .build();
+
+    build_test_externalities(config).execute_with(|| {
+        let _ = <Token as PalletToken<AccountId, Policy, IssuanceParams>>::reduce_patronage_rate_by(
+            token_id, decrement,
+        );
+
+        let approx = Token::token_info_by_id(token_id).patronage_info.rate;
+        let simm_diff = simmetric_diff!(approx.0.deconstruct(), expected.0.deconstruct());
+        assert!(simm_diff < 1_000u64); // accuracy: 1e-13 %
```

Comment:
Where does this value come from?

From my undestanding `simm_diff <= 3`, because `BlockRate::from_yearly_rate(x)` should be always accurate with +/- `1 / 10^18`.

Let's say you have: 
  - `initial_rate = BlockRate::from_yearly_rate(a)`,
  - `decrement_rate = BlockRate::from_yearly_rate(b)` and
  - `expected_rate = BlockRate::from_yearly_rate(c)` 

Now:
1. `initial_rate - decrement_rate` (let's call it `final_rate`) will be accurate with +/- `2 / 10^18`
2. `expected_rate` is accurate with +/- `1 / 10^18`
3. If you compare `final_rate` (+/- `2 / 10^18`) with `expected_rate` (+/- `1 / 10^18`), the max diff should be `3 / 10^18` and since all values are represented as "parts per quintillion" (which is 10^18), the max diff should be `3`.

Reply:
fix @ ea4f75c7df60c31f51d11e6cccef45f7e0ae6a80